### PR TITLE
fix(nginx): Fix proxy settings for Sentry

### LINKED ISSUE #447: CSRF failures after Nginx introduction
Recently updated my install, and I've had CSRF failures.

I believe the problem is the nginx relay in https://github.com/getsentry/onpremise/blob/master/nginx/nginx.conf#L44 (introduced in #421) is not passing the original Host header (at least that fixed it locally for me).

Log line from web:

> 16:46:05 [WARNING] django.security.csrf: Forbidden (Referer checking failed - https://sentry.example.com/auth/login/org/ does not match any trusted origins.): /auth/login/org/ (status_code=403 request=<WSGIRequest: POST u'/auth/login/org/'>)

### PR DESCRIPTION
**Pull request by BYK**
Fixes #447.

This patch should also fix issuer regarding large uploads such as minidumps by turning off any buffering.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Why would this fix it? We don’t do this in production.

File: nginx/nginx.conf [hunk 1]
```diff
@@ -42,6 +43,8 @@ http {
         }
         location / {
            proxy_pass http://sentry;
+           proxy_redirect off;
+           proxy_set_header Host       $host;
```
Comment (mattrobenolt) [inline nginx/nginx.conf:47]: Move this outside of the location block.

File: nginx/nginx.conf [hunk 2]
```diff
@@ -18,8 +18,9 @@ http {
                       '"$http_user_agent" "$http_x_forwarded_for"';
 
     sendfile        on;
-
     keepalive_timeout  65;
+    proxy_buffering off;
```
Comment (mattrobenolt) [inline nginx/nginx.conf:22]: These are not smart.

File: nginx/nginx.conf (hunk 1 above)
Comment (jan-auer) [reply nginx/nginx.conf:47]: See https://github.com/getsentry/sentry/blob/e8b7c215d9bb8fa789584ffc827611d060f9de23/config/reverse_proxy/nginx.conf#L49. I'll try to figure out how we missed this in the first place.

Comment (BYK) [reply nginx/nginx.conf:47]: Sure, didn't know `relay` cared about this.

**Pull request merged**